src/CMakeLists.txt

```diff
@@ -45,7 +45,12 @@ else()
     target_link_libraries(LIBTCOD PRIVATE ZLIB::ZLIB)
 endif()
 
-target_include_directories(LIBTCOD PRIVATE ./vendor)
+target_include_directories(LIBTCOD
+        PUBLIC
+            $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/libtcod/>
+            $<INSTALL_INTERFACE:libtcod/>
+        PRIVATE
+            ./vendor)
```

Comment:
The public interface should be the current directory `src/`, not `src/libtcod/`.

Reply:
No, you're wrong, the PUBLIC INTERFACE should point to where the .hpp/.h are - in this case `src/libtcod` and you'll ask why, and I'll tell you what it is because: when you use libtcod.hpp or libtcod.h you do it this way:

```cxx
#include "libtcod.hpp"
```

never in the following way:

```cxx
#include "src/libtcod.hpp"
```

Now, this problem exists, because libtcod does not follow modern recommendations for structuring project directories, but this is very easily solved.

Reply:
I apologize. I'm not very familiar with `BUILD_INTERFACE/INSTALL_INTERFACE` and the docs didn't help much.  I should have explained the 'why' more than the 'what'.

Libtcod used to be inconsistent about which directory counted as the base directory for the include files.  I had to pick either `src/libtcod/` or `src/` and I picked `src/` to reduce conflicts with other include files.  So when a program includes libtcod it should use `#include <libtcod.h>` (angle brackets because it isn't a relative path,) and that should include the header at `src/libtcod.h`.

The commands on how to install headers already exist in the `src/sources.cmake` script.  Headers install from `src/` to `${CMAKE_INSTALL_INCLUDEDIR}` and keep their relative directory.

On rereading the CMake docs my 2nd guess on how to set this up would be:
```cmake
# src/CMakeLists.txt
target_include_directories(LIBTCOD
    PUBLIC
        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
        $<INSTALL_INTERFACE:include/>
    PRIVATE
        ./vendor
)
```

I mostly use the [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html) for the layout of files and some other style choices.  I've had a hard time finding any kind of style guide for C/C++ in general.

Reply:
The way to structure the directories is as follows: https://cliutils.gitlab.io/modern-cmake/chapters/basics/structure.html

Reply:
> On rereading the CMake docs my 2nd guess on how to set this up would be

Again this is wrong, the INSTALL_INTERFACE defines what will be installed when the user does `sudo make install` and the BUILD_INTERFACE defines which headers the other targets will include.